src/QuadVelocityController.cpp

```diff
@@ -33,16 +35,20 @@ ran_once_(false)
 void QuadVelocityController::setTargetVelocity(geometry_msgs::Twist twist)
 {
     throttle_pid_.setSetpoint(twist.linear.z);
-    pitch_pid_.setSetpoint(twist.linear.x);
-    roll_pid_.setSetpoint(-twist.linear.y);
     yaw_pid_.setSetpoint(twist.angular.z);
+
+    // Pitch and roll velocities are transformed to vectors that pitch and roll can respond to
+    pitch_pid_.setSetpoint(twist.linear.x * std::cos(last_yaw_) + twist.linear.y * std::sin(last_yaw_));
+
+    // Invert roll because a positive y velocity means a negative roll by the right hand rule
```

Comment:
I think this should be fixed when feed forward is implemented. The PID loop should take care of the fact you use negative feedback to get a positive velocity.